### PR TITLE
[FIX]: Fix the traceback when click on the location and warehouse fil…

### PR DESCRIPTION
**Pull request by ppr-odoo**
Pad:https://pad.odoo.com/p/r.032e5b9719b7f74f02ff5ca5a1a7e6db
Task:https://www.odoo.com/web#id=1941639&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720

### CODE REVIEW

**Comment by mge-odoo** [comment]
Linked to https://github.com/odoo/odoo/pull/30368 where `self` in `context="{'field_id': self}"` is now a list of value (either integer for id, either string for search terms).

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -164,33 +165,37 @@ def _get_domain_locations(self):
...
-        if self.env.context.get('location', False):
-            if isinstance(self.env.context['location'], int):
-                location_ids = [self.env.context['location']]
-            elif isinstance(self.env.context['location'], str):
-                domain = [('complete_name', 'ilike', self.env.context['location'])]
-                if self.env.context.get('force_company', False):
-                    domain += [('company_id', '=', self.env.context['force_company'])]
-                location_ids = self.env['stock.location'].search(domain).ids
+
+        def _search_ids(model, value, force_company_id):
```
Comment (sle-odoo) [inline addons/stock/models/product.py]: values

File: addons/stock/models/product.py [hunk 2]
```diff
@@ -164,33 +165,37 @@ def _get_domain_locations(self):
...
+            ids = set()
+            domain = []
+            for item in value:
+                if isinstance(item, int):
+                    ids.add(item)
+                else:
+                    domain = expression.OR([[('name', 'ilike', item)], domain])
+            if force_company_id:
+                domain = expression.AND([[('company_id', '=', force_company_id)], domain])
```
Comment (sle-odoo) [inline addons/stock/models/product.py]: you also have to set the domains if only ids is set
```
if ids
    domain = ('id', 'in', list(ids))
```

File: addons/stock/models/product.py (hunk 1 above)
Comment (sle-odoo) [inline addons/stock/models/product.py]: add a comment: dummy field uset in the search view will always return a list of ids or a list of string

**Comment by sle-odoo** [comment]
robodoo r+

**Comment by sle-odoo** [comment]
robodoo r+

**Comment by KangOl** [comment]
@robodoo retry

**Pull request closed without merging**